File: python/leetcode/shuffle-an-array.py
class Solution:

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.orig = nums
        

    def reset(self):
        """
        Resets the array to its original configuration and return it.
        :rtype: List[int]
        """
        return self.orig
        

    def shuffle(self):
        """
        Returns a random shuffling of the array.
        :rtype: List[int]
        """
        options = self.orig.copy()
        l = len(options)
        shuf = []
        while l > 0:
            shuf.append(options.pop(int((random.random() * l) / 1)))
            l -= 1
        return shuf
        
        


# Your Solution object will be instantiated and called as such:
# obj = Solution(nums)
# param_1 = obj.reset()
# param_2 = obj.shuffle()

File: javascript/leetcode/shuffle-an-array.js
/**
 * @param {number[]} nums
 */
var Solution = function(nums) {
    this.orig = nums.slice(0)
    this.curr = nums.slice(0)
};

/**
 * Resets the array to its original configuration and return it.
 * @return {number[]}
 */
Solution.prototype.reset = function() {
    this.curr = this.orig.slice(0)
    return this.curr
};

/**
 * Returns a random shuffling of the array.
 * @return {number[]}
 */
Solution.prototype.shuffle = function() {
    options = this.orig.slice(0)
    l = options.length
    this.curr = []
    while (l > 0) {
        this.curr = this.curr.concat(options.splice(Math.floor(Math.random() * l), 1))
        l--
    }
    return this.curr
};

/** 
 * Your Solution object will be instantiated and called as such:
 * var obj = Object.create(Solution).createNew(nums)
 * var param_1 = obj.reset()
 * var param_2 = obj.shuffle()
 */
 